src/java.base/share/classes/jdk/internal/foreign/abi/fallback/LibFallback.java

```diff
@@ -43,7 +43,10 @@ private static boolean tryLoadLibrary() {
                     public Boolean run() {
                         try {
                             System.loadLibrary("fallbackLinker");
-                            init();
+                            if (!init()) {
+                                // library failed to initialize. Do not silently mark as unsupported
+                                throw new ExceptionInInitializerError("Fallback library failed to initialize");
+                            }
```

Comment:
It might be better to move this to after the try-catch, if init were to somehow fail with ULE then it suggests something else is wrong.